### PR TITLE
eliminated duplicately declared constants

### PR DESCRIPTION
**Pull request by simonjenga**
#94 eliminated duplicately declared constants

### CODE REVIEW

**Comment by dmarkov** [comment]
@simonjenga I will find a reviewer for your pull requests shortly, thanks for contribution!

**Comment by dmarkov** [comment]
@pinaf could you please review this PR

File: src/main/java/com/jcabi/http/JcabiHttp.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+/**
+ * This file contains constant definitions that are used within this project.
+ *
+ * Standard {@link Charset Charsets} defined here are guaranteed to be
+ * available on every implementation of the Java platform @since 1.7
+ * @see <a href="Charset#standard">Standard Charsets</a>
+ *
+ * @author Simon Njenga (simtuje@gmail.com)
+ * @version $Id$
+ * @since 0.10
```
Comment (pinaf) [inline src/main/java/com/jcabi/http/JcabiHttp.java]: @simonjenga 1.11

File: src/main/java/com/jcabi/http/JcabiHttp.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+ * @see <a href="http://http.jcabi.com">http://http.jcabi.com/</a>
+ */
+public final class JcabiHttp {
```
Comment (pinaf) [inline src/main/java/com/jcabi/http/JcabiHttp.java]: @simonjenga why not call this class `Constants` ?

File: src/main/java/com/jcabi/http/mock/MkGrizzlyAdapter.java [hunk 3]
```diff
@@ -222,17 +211,9 @@ public int queries() {
...
-            writer = new PrintWriter(
-                new OutputStreamWriter(
-                    response.getStream(),
-                    MkGrizzlyAdapter.ENCODING
-                )
-            );
-        } catch (final UnsupportedEncodingException ex) {
-            throw new IllegalStateException(ex);
-        }
+        final PrintWriter writer = new PrintWriter(
```
Comment (pinaf) [inline src/main/java/com/jcabi/http/mock/MkGrizzlyAdapter.java]: @simonjenga why remove this from the try block?

**Comment by pinaf** [comment]
@simonjenga 2 comments above

**Comment by pinaf** [comment]
@simonjenga I take it you removed some `try-catch` blocks because they cannot through the charset exception because you forced UTF-8?

**Comment by simonjenga** [comment]
@pinaf I had originally named the class ```Constants``` but i later saw that name is too commonly used by other libraries in the Java world and it wouldn't make much sense in our case. We need something unique that reflects our project. If you insist, i will name it so.

**Comment by pinaf** [comment]
@simonjenga multiple files have had their constructor javadoc changed... files that have nothing to do with this PR

**Comment by pinaf** [comment]
@simonjenga let's call it `Constant`

**Comment by simonjenga** [comment]
@pinaf from a grammatical point of view, this class should be named ```Constants``` since it contains a collection (many / more than one) of variables that should not be changed. Your suggestion of ```Constant``` refers to the ```singular``` form but for our case, we are referring to the ```plural```form  and hence ```Constants```.

**Comment by pinaf** [comment]
@simonjenga of course... the reason I suggested it is because you complained `Constants` was already present everywhere.

**Comment by pinaf** [comment]
@rultor merge

**Comment by yegor256** [comment]
@simonjenga I appreciate the work done here, but I have a concern. This "class" `Constants` is a violation of a basic principle of OOP: encapsulation. Now, our classes are not self-sufficient and depend on this new utility "class". Even though it looks convenient for a programmer, but it's against the OOP philosophy.

**Comment by pinaf** [comment]
@yegor256 so the original ticket idea was not good?

**Comment by yegor256** [comment]
@pinaf well, it was never approved as a "bug"... I think it's not good

**Comment by pinaf** [comment]
@yegor256 I see. I guess @simonjenga should close it then?

**Comment by yegor256** [comment]
@pinaf yes, I think so

**Comment by simonjenga** [comment]
@yegor256 @pinaf I think you are taking OOP too seriously and philosophical at the expense of engineering.

Some of these classes originally depended on an external third party library (apache Charsets/CharEncoding) to provide Charsets/UTF-8. What is now wrong with them depending on a class within the project?

This project also has dependencies on other libraries/jars which contain classes/utilities. Does that mean we are violating principles too?

What of this project grows and grows to be more than 50K lines of code? Does that mean we will continue duplicating these constants in the code over and over.

What is the use of classes if a single class can do everything on it's own.

http://en.wikipedia.org/wiki/Don%27t_repeat_yourself
http://en.wikipedia.org/wiki/Duplicate_code
http://en.wikipedia.org/wiki/Code_reuse
http://c2.com/cgi/wiki?OnceAndOnlyOnce
http://www.c2.com/cgi/wiki?DontRepeatYourself

**Comment by yegor256** [comment]
@simonjenga we definitely need to try to do as much as possible in order to get rid of code duplication. But not by moving constants into one place. We need to introduce new classes that will encapsulate the functionality that is being repeated. Let's say you want to do `new String(byteArray, "UTF-8")` and you don't want to have this `"UTF-8"` literal in many places. Introduce a new class and do: `new UtfString(byteArray).toString()`. See my point?

And about Apache and other libraries - they violate all principles of OOP. No need to use them as examples. Most of them are just horrible.

**Comment by pinaf** [comment]
@simonjenga as you can see @yegor256 is very strict about this issue. it is something he talks a lot about in his blog and webinars and I agree with him on this point - we should think about objects and not procedurally. But I confess it is hard work untwisting the mind!

**Comment by simonjenga** [comment]
@pinaf While you are at it reading the blogs and listening to the webinars, i would also recommend you the topic on "Perfectionism ". Your friend google will help you in finding the relevant links.

**Comment by pinaf** [comment]
@simonjenga It was never my intention to upset you or to pontificate and I'm sorry if that's the way my message sounded. I was simply expressing my opinion on the matter.

**Comment by simonjenga** [comment]
@pinaf You didn't upset me. No need to feel any guilt/remorse. I only wanted to provide you with something you can be able to compare with. You now have two sides of the coin with you to make a comparison. While becoming better and more skilled is good, it should be done with caution! :)

**Comment by pinaf** [comment]
@simonjenga great. I guess let's close this PR.

**Comment by pinaf** [comment]
@simonjenga @yegor256 do you mind closing this?

**Comment by pinaf** [comment]
@yegor256 please close this issue since @simonjenga clearly doesn't want to.

**Comment by yegor256** [comment]
@pinaf closed

**Comment by dmarkov** [comment]
@pinaf Thanks a lot, I just topped your account for **28 mins**, transaction ID `AP-1Y956564F39725837`, total time was 479 hours and 55 mins... extra minutes for review comments (c=13)... +28 to your rating, your total score is [+6615](http://www.netbout.com/b/35128?open=rating)

**Comment by simonjenga** [comment]
Apologies for any inconvenience caused.

**Pull request closed without merging**